cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/updatetemplates/UpdateTemplatesDialog.java

```diff
@@ -0,0 +1,130 @@
+package com.devonfw.cobigen.eclipse.updatetemplates;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.UUID;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import com.devonfw.cobigen.api.exception.CobiGenRuntimeException;
+import com.devonfw.cobigen.eclipse.common.constants.InfrastructureConstants;
+import com.devonfw.cobigen.eclipse.common.constants.external.CobiGenDialogConstants.UpdateTemplateDialogs;
+import com.devonfw.cobigen.eclipse.common.tools.ResourcesPluginUtil;
+
+/**
+ * Dialog to update templates from Maven central
+ */
+public class UpdateTemplatesDialog extends Dialog {
+
+    /** Logger instance. */
+    private static final Logger LOG = LoggerFactory.getLogger(UpdateTemplatesDialog.class);
+
+    /**
+     * Creates a new {@link UpdateTemplatesDialog}
+     */
+    public UpdateTemplatesDialog() {
+        super(Display.getDefault().getActiveShell());
+
+        // Create a new trust manager that trust all certificates
+        TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+            @Override
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+
+            @Override
+            public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+            }
+
+            @Override
+            public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+            }
+        } };
+
+        // Activate the new trust manager
+        try {
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+        } catch (Exception e) {
```

Comment:
Please never just catch all exceptions and ignore it. Especially not in this case.
Either log or throw an exception.